plugins/action/system/shutdown.py

```diff
@@ -0,0 +1,234 @@
+# Copyright: (c) 2016-2018, Matt Davis <mdavis@ansible.com>
+# Copyright: (c) 2018, Sam Doran <sdoran@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import random
+import time
+
+from datetime import datetime, timedelta
+
+from ansible.errors import AnsibleError, AnsibleConnectionFailure
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.common.collections import is_string
+from ansible.module_utils.common.validation import check_type_str
+from ansible.plugins.action import ActionBase
+from ansible.utils.display import Display
+
+display = Display()
+
+
+class TimedOutException(Exception):
+    pass
+
+
+class ActionModule(ActionBase):
+    TRANSFERS_FILES = False
+    _VALID_ARGS = frozenset((
+        'connect_timeout',
+        'msg',
+        'delay',
+        'search_paths'
+    ))
+
+    DEFAULT_CONNECT_TIMEOUT = None
+    DEFAULT_PRE_SHUTDOWN_DELAY = 0
+    DEFAULT_SHUTDOWN_MESSAGE = 'Shut down initiated by Ansible'
+    DEFAULT_SHUTDOWN_COMMAND = 'shutdown'
+    DEFAULT_SHUTDOWN_COMMAND_ARGS = '-h {delay_min} "{message}"'
+    DEFAULT_SUDOABLE = True
+
+    DEPRECATED_ARGS = {}
```

Comment:
```suggestion
```

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/700/commits/a156e49f426643ea8235136b909a9180cafa8ac6.